test/conversion_test.go

```diff
@@ -947,3 +1136,162 @@ func TestPipelineRunCRDConversion(t *testing.T) {
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	repo := fmt.Sprintf("%s:5000/tektonbundlessimple", getRegistryServiceIP(ctx, t, c, namespace))
+	ref, err := name.ParseReference(repo)
+	if err != nil {
+		t.Fatalf("Failed to parse %s as an OCI reference: %s", repo, err)
+	}
+
+	taskName := helpers.ObjectNameForTest(t)
+	pipelineName := helpers.ObjectNameForTest(t)
+	v1beta1TaskRunName := helpers.ObjectNameForTest(t)
+	task := parse.MustParseV1beta1Task(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  steps:
+  - name: hello
+    image: alpine
+    script: 'echo Hello'
+`, taskName, namespace))
+	pipeline := parse.MustParseV1beta1Pipeline(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  tasks:
+  - name: hello-world
+    taskRef:
+      resolver: bundles
+      params:
+      - name: bundle
+        value: %s
+      - name: name
+        value: %s
+`, pipelineName, namespace, repo, taskName))
+	v1beta1TaskRun := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(v1beta1TaskRunWithBundleYaml, v1beta1TaskRunName, namespace, taskName, repo))
+	v1TaskRunExpected := parse.MustParseV1TaskRun(t, fmt.Sprintf(v1TaskRunWithBundleExpectedYaml, v1beta1TaskRunName, namespace, repo, taskName, v1beta1TaskRunName))
+	v1beta1TaskRunRoundTripExpected := parse.MustParseV1beta1TaskRun(t, fmt.Sprintf(v1beta1TaskRunWithBundleRoundTripYaml, v1beta1TaskRunName, namespace, repo, taskName, v1beta1TaskRunName))
+
+	// Write the task and pipeline into an image to the registry in the proper format.
+	rawTask, err := yaml.Marshal(task)
+	if err != nil {
+		t.Fatalf("Failed to marshal task to yaml: %s", err)
+	}
+
+	rawPipeline, err := yaml.Marshal(pipeline)
+	if err != nil {
+		t.Fatalf("Failed to marshal task to yaml: %s", err)
+	}
+	img := empty.Image
+	taskLayer, err := tarball.LayerFromReader(bytes.NewBuffer(rawTask))
+	if err != nil {
+		t.Fatalf("Failed to create oci layer from task: %s", err)
+	}
+	pipelineLayer, err := tarball.LayerFromReader(bytes.NewBuffer(rawPipeline))
+	if err != nil {
+		t.Fatalf("Failed to create oci layer from pipeline: %s", err)
+	}
+	img, err = mutate.Append(img, mutate.Addendum{
+		Layer: taskLayer,
+		Annotations: map[string]string{
+			"dev.tekton.image.name":       taskName,
+			"dev.tekton.image.kind":       strings.ToLower(task.Kind),
+			"dev.tekton.image.apiVersion": task.APIVersion,
+		},
+	}, mutate.Addendum{
+		Layer: pipelineLayer,
+		Annotations: map[string]string{
+			"dev.tekton.image.name":       pipelineName,
+			"dev.tekton.image.kind":       strings.ToLower(pipeline.Kind),
+			"dev.tekton.image.apiVersion": pipeline.APIVersion,
+		},
+	})
+	if err != nil {
+		t.Fatalf("Failed to create an oci image from the task and pipeline layers: %s", err)
+	}
+
+	// Publish this image to the in-cluster registry.
+	publishImg(ctx, t, c, namespace, img, ref)
+
+	v1TaskRunExpected.Status.Provenance = &v1.Provenance{
```

Comment:
maybe add some comments for the below code sections? It's a bit long and they may help to understand which part is testing what